File: frontend/src/types/room.ts
export type RoomCategory = "Premium" | "Standard" | "Haut de Gamme";

export type RoomType = "Petite" | "Moyenne" | "Grande" | "Hangar" | "Parking" | "Espace Atypique";

export type RoomAmenity = {
  icon: string;
  name: string;
  description?: string;
};

export type RoomService = {
  icon: string;
  name: string;
  description: string;
  includedInPrice: boolean;
  priceIfExtra?: number;
};

export type RoomImage = {
  src: string;
  alt: string;
  featured?: boolean;
};

export type Room = {
  id: string;
  name: string;
  slug: string;
  description: string;
  shortDescription: string;
  category: RoomCategory;
  type: RoomType;
  capacity: {
    min: number;
    max: number;
    optimal: number;
  };
  size: number; // in square meters
  pricePerHour: number;
  pricePerDay: number;
  location: {
    address: string;
    city: string;
    postalCode: string;
    country: string;
    coordinates: {
      lat: number;
      lng: number;
    };
  };
  amenities: RoomAmenity[];
  services: RoomService[];
  images: RoomImage[];
  availabilityConfirmationRequired: boolean;
  rating?: number;
  reviews?: number;
};

File: backend/app/schemas/room.py
from pydantic import BaseModel, Field
from typing import List, Dict, Optional, Literal

# Correspondant aux types de frontend
RoomCategory = Literal["Premium", "Standard", "Haut de Gamme"]
RoomType = Literal["Petite", "Moyenne", "Grande", "Hangar", "Parking", "Espace Atypique"]

class RoomAmenity(BaseModel):
    icon: str
    name: str
    description: Optional[str] = None

class RoomService(BaseModel):
    icon: str
    name: str
    description: str
    includedInPrice: bool
    priceIfExtra: Optional[float] = None

class RoomImage(BaseModel):
    src: str
    alt: str
    featured: Optional[bool] = None

class RoomCapacity(BaseModel):
    min: int
    max: int
    optimal: int

class RoomLocation(BaseModel):
    address: str
    city: str
    postalCode: str
    country: str
    coordinates: Dict[str, float]

class RoomCreate(BaseModel):
    name: str
    slug: str
    description: str
    shortDescription: str
    category: RoomCategory
    type: RoomType
    capacity: RoomCapacity
    size: float
    pricePerHour: float
    pricePerDay: float
    location: RoomLocation
    amenities: List[RoomAmenity]
    services: List[RoomService]
    images: List[RoomImage]
    availabilityConfirmationRequired: bool = False
    rating: Optional[float] = None
    reviews: Optional[int] = None

class RoomResponse(RoomCreate):
    id: str